extensions/src/main/java/org/pentaho/platform/web/http/filters/ProxyTrustingFilter.java

```diff
@@ -433,4 +469,15 @@ protected void becomeUser( final String principalName ) {
   protected void setSystemLocaleOverrideCode( String localeOverrideCode ) {
     LocaleHelper.setThreadLocaleOverride( LocaleHelper.parseLocale( localeOverrideCode ) );
   }
+
+  private void doCsrfValidation( HttpServletRequest request ) throws AuthenticationException, ServletException, IOException {
+    if ( csrfValidator != null ) {
+      try {
+        csrfValidator.validateRequestOfMutationOperation( request,this.getClass(), CSRF_OPERATION_NAME );
+      } catch ( AccessDeniedException ex ) {
+        throw new CsrfValidationAuthenticationException( ex );
+      }
+    }
+  }
```

Comment:
In this context, the `AccessDeniedException` should be logged and then converted into calling:
```java
response.sendError( HttpStatus.SC_FORBIDDEN );
```

`chain.doFilter(...)` should not be called in this case.

The ServletException and IOException exceptions are being correctly handled — should be thrown to the caller of `doFilter`.

Reply:
addressed review comments